service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DiaryPlanStatementMapper.java

```diff
@@ -0,0 +1,147 @@
+            .availabilityTime(buildAvailabilityTime(procedureRequest));
+
+        buildEffectiveTime(procedureRequest).map(builder::effectiveTime);
+        buildText(procedureRequest).map(builder::text);
+
+        return TemplateUtils.fillTemplate(PLAN_STATEMENT_TEMPLATE, builder.build());
+    }
+
+    private Optional<String> buildEffectiveTime(ProcedureRequest procedureRequest) {
+        Date date = null;
+        if (procedureRequest.hasOccurrenceDateTimeType()) {
+            date = procedureRequest.getOccurrenceDateTimeType().getValue();
+        } else if (procedureRequest.hasOccurrencePeriod()) {
+            Period occurrencePeriod = procedureRequest.getOccurrencePeriod();
+            date = occurrencePeriod.hasEnd() ? occurrencePeriod.getEnd() : occurrencePeriod.getStart();
+        }
+
+        return Optional.of(formatEffectiveDate(date));
+    }
+
+    private String formatEffectiveDate(Date date) {
+        return date != null ? String.format(FULL_DATE, DateFormatUtil.formatDate(date)) : EMPTY_DATE;
+    }
+
+    private String buildAvailabilityTime(ProcedureRequest procedureRequest) {
+        if (procedureRequest.hasAuthoredOn()) {
+            return DateFormatUtil.formatDate(procedureRequest.getAuthoredOn());
+        }
+
+        throw new EhrMapperException(
+            String.format("ProcedureRequest: %s does not have required authoredOn", procedureRequest.getId())
+        );
+    }
+
+    private Optional<String> buildText(ProcedureRequest procedureRequest) {
+        return Stream.of(
+            getRequester(procedureRequest),
+            getReasonCode(procedureRequest),
+            getNotes(procedureRequest)
+        )
+            .filter(Optional::isPresent)
+            .map(Optional::get)
+            .reduce(joining());
+    }
+
+    private Optional<String> getReasonCode(ProcedureRequest procedureRequest) {
+        return procedureRequest.getReasonCode()
+            .stream()
+            .map(reasonCode -> {
+                if (StringUtils.isNoneBlank(reasonCode.getText())) {
+                    return reasonCode.getText();
+                }
+                return getDisplay(reasonCode);
+            })
+            .filter(StringUtils::isNoneBlank)
+            .reduce(joining());
+    }
+
+    private String getDisplay(org.hl7.fhir.dstu3.model.CodeableConcept reasonCode) {
+        return reasonCode.getCoding()
+            .stream()
+            .map(Coding::getDisplay)
+            .reduce(joining()).orElse(StringUtils.EMPTY);
+    }
+
+    private Optional<String> getNotes(ProcedureRequest procedureRequest) {
+        return procedureRequest.getNote()
+            .stream()
+            .map(Annotation::getText)
+            .reduce(joining());
+    }
+
+    private BinaryOperator<String> joining() {
+        return (value1, value2) -> value1 + COMMA + value2;
+    }
+
+    private Optional<String> getRequester(ProcedureRequest procedureRequest) {
```

Comment:
I don't think this produces the specified output, but I can see how the spreadsheet may be confusing.

> requester.agent.reference is Organisation: 
> form a string along the lines of "Recall Organisation: { Organisation.name} " and prepend it to the <text/> element

Means you will need to find the Organisation resource in the bundle referred to by `requester.agent.reference` and get its name.

You need to then build the string "Recall Organisation: { Organisation.name} " and prepend that to the text. 

As written this will just prepend the "display" property of the reference. This property is not used by GP Connect.